src/View/Helper/PaginatorHelper.php

```diff
@@ -1169,4 +1170,41 @@ public function implementedEvents()
     {
         return [];
     }
+
+    /**
+     * dropdown array for Select limit
+     *
+     * @param array $limits This is array of options.
+     * @param int|null $default Default limit for option selecting. Default value is $this->param('perPage').
+     * @param array $options Options for Select tag attributes like class, id or event
+     *
+     * @return string html output.
+     */
+    public function limitControl(array $limits = [], $default = null, array $options = [])
+    {
+        $out = $this->Form->create(null, ['type' => 'get']);
+
+        if (empty($default) || !is_numeric($default)) {
+            $default = $this->param('perPage');
+        }
+
+        if (empty($limits)) {
+            $limits = [
+                '20' => '20',
+                '50' => '50',
+                '100' => '100'
+            ];
+        }
+
+        if (!in_array($default, $limits)) {
+            $limits += [$default => $default];
+        }
+
+        ksort($limits);
+
+        $out .= $this->Form->control('limit', ($options + ['type' => 'select', 'label' => __('View'), 'value' => $default, 'options' => $limits, 'onChange' => 'this.form.submit()']));
```

Comment:
Line too long.
What's with that bracket before $options?
May be put that into a separate assignment before assembling the output.